Tutorial: Add Working with SurrealDB over HTTP via Postman tutorial (#297)
surrealdb/docs.surrealdb.com: versioned_docs/version-1.1.x/tutorials/working-with-surrealdb-over-http-via-postman.mdx

Reviewer: Sometimes the headings are title case, sometimes sentence case, would be good to standardise on one way of doing it.

Author: I think we can use the Title case for headers.

Author: Updated the titles to be title case

---
sidebar_position: 2
---

# Working with SurrealDB over HTTP via Postman

# Working with SurrealDB over HTTP via Postman

SurrealDB provides a RESTful HTTP API for interacting with the database. This API is designed to be simple and easy to use, and is the primary way to interact with SurrealDB.

In this tutorial, you will learn how to query SurrealDB endpoints via the [Postman collection](https://postman.com/surrealdb/workspace/surrealdb/collection/19100500-3da237f3-588b-4252-8882-6d487c11116a). SurrealDB supports requests via HTTP & Rest with which you can handle different queries from creating simple tables to having Graph relationships between complex tables.

Check out the [HTTP & Rest integration documentation](/docs/integration/http) for a detailed list of all the endpoints supported.

## Prerequisites

This tutorial assumes that you have the following: 

- A Postman account to fork the [SurrealDB collection](https://postman.com/surrealdb/workspace/surrealdb/collection/19100500-3da237f3-588b-4252-8882-6d487c11116a)
- SurrealDB installed - If you do not, see the [installation guide](/docs/installation/overview) specific to your machine.

## Getting started.

> Before heading forking the Collection from Postman, ensure that you have your SurrealDB instance running locally because by default the collection endpoint is set to [`localhost:8000`](http://localhost:8000) or [`http://127.0.0.1:8000`](http://127.0.0.1:8000/)

Reviewer: ```suggestion
Before forking the Collection from Postman, ensure that you have your SurrealDB instance running locally because by default the collection endpoint is set to [`localhost:8000`](http://localhost:8000) or [`http://127.0.0.1:8000`](http://127.0.0.1:8000/)
```

To do so run the [Start command](/docs/cli/start) in your terminal: 

```sql
surreal start --auth --user root --pass root
```

> The above command [enables authentication](/docs/cli/start#authentication) with `--auth` and specifies that the user and password are `root` - Note that this isn’t the permission and can be replaced with any name.  

Reviewer: ```suggestion
The above command [enables authentication](/docs/cli/start#authentication) with `--auth` and specifies that the user and password are `root` - Note that this is just for demonstration purposes. You can replace it with any other word.
```

It is also important to note that the `namespace` and `database` values for the Postman collection are set to `test` by default. 

![Info-statement](/img/Info-statement.png)

You can see the JSON output using the [INFO statement](/docs/surrealql/statements/info) in the body of the request 

```sql
INFO FOR ROOT / DB / NS
```

Below is the output of querying for the info of the `Namespace` you can get this output by running 

```sql
INFO FOR NS
```
![Info-Output](/img/Info-Output.png)

## Defining a Scope User

To define a scope user, first navigate to the `POST /sql`  endpoint and in the header of the request add the following fields:

```json
Accept:application/json
NS:{{namespace}}
DB:{{database}}
```
You can also [define a user](/docs/surrealql/statements/define/user) with other credentials such as Root or Database user. For this tutorial, we will be using only Scope users.

:::note
__Note:__ The Namespace and Database fields are set to the value `test` by default. You can change this in the collection settings.
:::

> In the request body, [Define the scope](/docs/surrealql/statements/define/scope) `human` with a session of 24 hours: 

Reviewer: ```suggestion
In the request body, [define the scope](/docs/surrealql/statements/define/scope) `human` with a session of 24 hours: 
```
